src/ucp/core/ucp_mm.c

```diff
@@ -1385,16 +1400,55 @@ ucs_status_t ucp_mem_reg_md_map_update(ucp_context_h context)
     context->alloc_md_index[alloc_mem_type] = alloc_memh->alloc_md_index;
     ucp_memh_put(context, alloc_memh);
 
-    status = ucp_memh_register(context, &reg_memh.memh,
-                               context->reg_md_map[alloc_mem_type], &buff,
-                               sizeof(buff), alloc_mem_type,
+    return UCS_OK;
+}
+
+static void ucp_mem_reg_md_map_exclude_cuda(ucp_context_h context,
+                                            ucp_md_map_t reg_md_map,
+                                            ucp_md_map_t *md_map_included,
+                                            ucp_md_map_t *md_map_excluded)
+{
+    int i;
+    ucp_tl_resource_desc_t *rsc;
+
+    for (i = 0; i < context->num_tls; ++i) {
+        rsc = &context->tl_rscs[i];
+        if (rsc->tl_rsc.dev_type == UCT_DEVICE_TYPE_ACC) {
+            *md_map_excluded |= UCS_BIT(rsc->md_index);
+        }
+    }
+
+    *md_map_excluded &= reg_md_map;
+    *md_map_included  = reg_md_map & ~(*md_map_excluded);
+}
+
+ucs_status_t ucp_mem_reg_md_map_update(ucp_context_h context)
+{
+    const ucs_memory_type_t reg_mem_type = UCS_MEMORY_TYPE_HOST;
+    UCS_STRING_BUFFER_ONSTACK(strb, 256);
+    ucp_mem_dummy_handle_t reg_memh = {
+        .memh.alloc_md_index = UCP_NULL_RESOURCE
+    };
+    ucp_md_map_t md_map_to_reg = 0, md_map_excluded = 0;
+    ucs_memory_type_t mem_type;
+    ucp_md_index_t md_index;
+    ucs_status_t status;
+    uint8_t buff;
+
+    /* Include to md_map_to_reg only mds that don't support registration on
+     * CUDA memory type to avoid calling CUDA API in frames of UCP context
+     * creation. */
+    ucp_mem_reg_md_map_exclude_cuda(context, context->reg_md_map[reg_mem_type],
+                                    &md_map_to_reg, &md_map_excluded);
+    status = ucp_memh_register(context, &reg_memh.memh, md_map_to_reg, &buff,
+                               sizeof(buff), reg_mem_type,
```

Comment:
i think we want to avoid cuda registration, so we have to pass excluded map here

Reply:
`md_map_excluded` contains memory domains corresponding to resources which device type is accelerator.
`md_map_to_reg` contains `context->reg_md_map[reg_mem_type]` excluding `md_map_excluded`.
Thus we pass `md_map_to_reg` to testing memory register procedure to avoid call memory registration for memory domains from `md_map_excluded`.

Reply:
Updated variable names.